### ISSUE TITLE
Add end flag to data downloads

### ISSUE THREAD

**Issue by 153957** [HiSPARC/publicdb#127]
Add a comment line at the end of data downloads to indicate successful download.

e.g. to indicate that that the requested date range was entirely traversed
```
# end reached
```
Currently a `# No coincidences were found..` is shown if no coincidences are found but the entire range is traversed.
If these comments are not present when a download finishes it may indicate that an error occurred.
In that case it might be good to retry downloading from the last coincidence that was received.

**Comment by davidfokkema** [comment]
I love the simplicity of that idea!

**Comment by tomkooij** [comment]
+1 This is indeed a very useful enhancement. 
It will allow us to build a much more robust sapphire.download_coincidences().

**Comment by 153957** [comment]
The problem with simply (re)starting the download at the timestamp + 1 (n)s of the last retrieved coincidence may be that you will encounter the same error/exception that caused the previous fail..
e.g. a corrupt file, or missing station data..
It may be wise to skip the entire day in which it failed, but that may again leave the user with incomplete data.. as long as the errors/warnings are clear in communicating what SAPPHiRE is doing to the user it should be fine.. for example;
```
start download...
[progressbar]
Downloading failed at [datetime], restarting download for [datetime]-[end]
[progressbar]
Downloading failed at [datetime], not retrying.
```

**Comment by tomkooij** [comment]
I agree. If we implement an exit code in sapphire.download_coincidences() we can script around failures. (Or at least we can download many coincidences automatically and be informed about  success/failures).

**Comment by 153957** [comment]
Fixed in #128
